specifyweb/frontend/js_src/lib/components/QueryBuilder/Export.tsx

```diff
@@ -59,6 +65,33 @@ export function QueryExportButtons({
     });
   }
 
+  const [separator] = userPreferences.use(
+    'queryBuilder',
+    'behavior',
+    'exportFileDelimiter'
+  );
+
+  const selectedResults = resultsArray?.filter((item) => {
+    if (item && typeof item[0] === 'number') {
```

Comment:
please avoid implicit boolean conversion (`item && `)
this should have been written simpler as `typeof item?.[0] === 'number'`)

Reply:
checking for first item being a number is a potential bug - if first query column is a numeric field, this will fail
you should instead check if query is distinct (using queryResource - see other usages for example)